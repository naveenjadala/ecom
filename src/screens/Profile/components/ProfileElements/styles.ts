import {StyleSheet} from 'react-native';
import {colors} from '../../../../assets/colors';
import {getFontSize} from '../../../../assets/fonts/FontSizes';

export const eventStyles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {color: colors.text, fontSize: getFontSize(12), paddingVertical: 10},
});
