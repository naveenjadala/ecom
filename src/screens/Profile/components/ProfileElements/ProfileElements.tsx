import {StyleSheet, Text, TouchableOpacity} from 'react-native';
import React, {FC} from 'react';
import {eventStyles} from './styles';

type ProfileElementProps = {
  title: string;
  elementImage?: JSX.Element;
  onClick: () => void;
};

/**
 * React functional component for displaying profile events.
 *
 * @param {ProfileElementProps} title - The title of the profile event.
 * @param {ProfileElementProps} elementImage - The image element for the profile event.
 * @return {JSX.Element} The rendered profile events component.
 */
const ProfileElements: FC<ProfileElementProps> = ({
  title,
  elementImage,
  onClick,
}) => {
  return (
    <TouchableOpacity
      accessibilityRole="button"
      onPress={() => {
        onClick();
      }}
      style={eventStyles.container}>
      {elementImage}
      <Text style={eventStyles.text}>{title}</Text>
    </TouchableOpacity>
  );
};

export default ProfileElements;

const styles = StyleSheet.create({});
