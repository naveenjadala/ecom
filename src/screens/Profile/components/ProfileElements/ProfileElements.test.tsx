import {render, screen} from '../../../../test/test-utils';
import ProfileElements from './ProfileElements';

test('should render ProfileEvents snapShort', () => {
  render(
    <ProfileElements
      title="ProfileEvents"
      onClick={function (): void {
        throw new Error('Function not implemented.');
      }}
    />,
  );
  expect(screen).toMatchSnapshot();
});

describe('should render ProfileEvents element', () => {
  test('should render touchable element', () => {
    render(
      <ProfileElements
        title="ProfileEvents"
        onClick={function (): void {
          throw new Error('Function not implemented.');
        }}
      />,
    );
    const getClickableElementView = screen.getByRole('button');
    expect(getClickableElementView).toBeTruthy();
  });

  test('should render text element', () => {
    render(
      <ProfileElements
        title="ProfileEvents"
        onClick={function (): void {
          throw new Error('Function not implemented.');
        }}
      />,
    );
    const getTextView = screen.getByText('ProfileEvents');
    expect(getTextView).toBeTruthy();
  });

  //   test('should render text element', () => {
  //     render(<ProfileEvents image="ProfileEvents" />);
  //     const getTextView = screen.getByText('ProfileEvents');
  //     expect(getTextView).toBeTruthy();
  //   });
});
