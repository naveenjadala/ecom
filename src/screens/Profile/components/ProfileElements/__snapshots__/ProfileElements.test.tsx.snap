// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProfileEvents snapShort 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "minHeight": "100%",
      "minWidth": "100%",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 6,
          "paddingVertical": 10,
        }
      }
    >
      ProfileEvents
    </Text>
  </View>
</RCTSafeAreaView>
`;
